lib/r10k/action/deploy/environment.rb

```diff
@@ -65,7 +65,7 @@ def visit_deployment(deployment)
         ensure
           if (postcmd = @settings[:postrun])
             if postcmd.grep('$modifiedenvs').any?
-              envs = deployment.environments.map { |e| e.name }
+              envs = deployment.environments.map { |e| e.dirname }
               envs.reject! { |e| !@argv.include?(e) } if @argv.any?
```

Comment:
`@argv` already has non word chars converted to underscores here. https://github.com/puppetlabs/r10k/blob/c699bd3ba0f9bf2f1d049672fad9a1b85355fee3/lib/r10k/action/deploy/environment.rb#L27